src/lib.rs

```diff
@@ -670,6 +672,35 @@ impl Library {
                 _ => (),
             }
         }
+
+        let mut linker_options = words.iter().filter(|arg| arg.starts_with("-Wl,"));
+        while let Some(option) = linker_options.next() {
+            let mut pop = false;
+            let mut ld_option = vec![];
+            for subopt in option[4..].split(',') {
+                if pop {
+                    pop = false;
+                    continue;
+                }
+
+                match subopt {
+                    "-framework" | "-isystem" | "-iquote" | "idirafter" => {
```

Comment:
The `-iXXX` ones are not linker flags and AFAIU can't be found as `-Wl,-isystem` etc

Reply:
And maybe it would make sense to skip all `-Wl` in the loop above and then handle `-framework` directly here

Reply:
Gotcha, this makes sense.  I just kind of mechanically did the translation, not really thinking.

Reply:
The moving of `-framework` to this loop sounds good at first but this loop doesn't handle bare arguments it only handles arguments with `-Wl,` prefixed on them.  looking at some stackoverflow answers it seems like this can be normal so something tells me we should leave `-framework` where it is.

Reply:
It's e.g. `-Wl,-framework,Cocoa` in the GTK pkg-config files instead of plain `-framework`

Reply:
I.e. what I mean is that the strange `-Wl` special-casing in the loop above could potentially go away, and instead your new loop could handle all the `-Wl` arguments in one go and special case `-framework` there.

Reply:
On a second thought, let's just keep it like this.